envoy/config/filter/http/health_check/v2/health_check.proto

```diff
@@ -30,4 +31,10 @@ message HealthCheck {
   // names and the minimum percentage of servers in each of those clusters that
   // must be healthy in order for the filter to return a 200.
   map<string, envoy.type.Percent> cluster_min_healthy_percentages = 4;
+
+  // [#not-implemented-hide:]
+  // Specifies a set of health check request headers to match on. The health check filter will
+  // check a request’s headers against all the specified headers. A request will be considered a
+  // health check request if all headers are present.
```

Comment:
Can you specify how this relates to the "endpoint" parameter? (In a perfect world we would deprecate "endpoint" and just have people match on ":path" so might consider that also.)

Reply:
Good point. As it's written, :path must match both "endpoint" and headers if specified.

I don't think we plan on deprecating "endpoint" in the PR that will implement header matching. Should I open an issue?